File: lista3/Figury.cpp
#include "Figury.hpp"
#include "Kolo.hpp"
#include "Szesciokat.hpp"
#include "Pieciokat.hpp"
#include "Kwadrat.hpp"
#include "Prostokat.hpp"
#include "Romb.hpp"

Figura::Figura() {}
Figura::~Figura() {}
Czworokat::Czworokat() {}
Czworokat::~Czworokat() {}

Kolo::Kolo() {
    double Promien;
}
Kolo::~Kolo(){}
double Kolo::obwod() {
    return 2*Promien*M_PI;
}
double Kolo::pole() {
    return M_PI*Promien*Promien;
}
Szesciokat::Szesciokat() {
    double bok;
}
Szesciokat::~Szesciokat() {}
double Szesciokat::obwod() {
    return 6*bok;
}
double Szesciokat::pole() {
    return 1.5*sqrt(3)*bok*bok;
}
Pieciokat::Pieciokat() {
    double bok;
}
Pieciokat::~Pieciokat(){}
double Pieciokat::obwod() {
    return 5*bok;
}
double Pieciokat::pole() {
    return 0.25*bok*bok*sqrt(25+10*sqrt(5));
}
Kwadrat::Kwadrat() {}
Kwadrat::~Kwadrat() {}
double Kwadrat::obwod() {
    return 4*bok1;
}
double Kwadrat::pole() {
    return bok1*bok1;
}
Prostokat::Prostokat() {}
Prostokat::~Prostokat() {}
double Prostokat::obwod() {
    return bok1+bok2+bok3+bok4;
}
double Prostokat::pole() {
    return std::min(bok1,bok2)*std::max(bok3,bok4);
}
Romb::Romb() {}
Romb::~Romb() {}
double Romb::obwod() {
    return 4*bok1;
}
double Romb::pole() {
    return bok1*bok1*sin(kat);
}

File: lista3_enumy/Figury.java
interface JednoArgumentowe {
    public double pole(double arg);
    public double obwod(double arg);
}
interface DwuArgumentowe {
    public double pole(double arg1, double arg2);
    public double obwod(double arg1, double arg2);
}

public class Figury {
    public enum JednoParametrowe implements JednoArgumentowe{
        KOLO {
            public double pole (double arg){
                return 2*arg*Math.PI;
            }
            public double obwod (double arg){
                return Math.PI*arg*arg;
            }
        },
        PIECIOKAT {
            public double obwod(double arg) {
                return 5*arg;
            }
            public double pole(double arg) {
                return 0.25*arg*arg*Math.sqrt(25+10*Math.sqrt(5));
            }
        },
        SZESCIOKAT {
            public double obwod(double arg){
                return 6*arg;
            }
            public double pole(double arg) {
                return 1.5*Math.sqrt(3)*arg*arg;
            }
        },
        KWADRAT {
            public double obwod(double arg){
                return 4*arg;
            }
            public double pole(double arg) {
                return arg*arg;
            }
        };
    }
    public enum DwuParametrowe implements DwuArgumentowe{
        PROSTOKAT {
            public double pole(double arg1, double arg2){
                return arg1*arg2;
            }
            public double obwod(double arg1, double arg2){
                return 2*arg1+2*arg2;
            }
        },
        ROMB {
            public double pole (double arg1, double arg2){
                return arg1*arg1*Math.sin(Math.toRadians(arg2));
            }
            public double obwod (double arg1, double arg2){
                return 4*arg1;
            }
        };
    }
}

